protocol/0014-ORDT-order_types.md

```diff
@@ -46,9 +46,129 @@ Notes on scope of current version of this spec:
+
+* When `displayed quantity < minimum peak size` and `remaining > displayed quantity` the order will be refreshed:
+
+  * The refresh happens at the end of the transaction when the order becomes eligible for refresh.
+
+  * On refresh `display quantity` is set to `min(remaining, initial peak size)`.
+
+  * A refresh simulates a cancel/replace, which means that on refresh an iceberg order will always lose time priority relative to other orders at the same price.
+
+  * If multiple iceberg orders need to be refreshed at the same time, they are refreshed in the order that their eligibility for refresh was triggered, so the iceberg that dropped below its `minimum peak size` first is refreshed first (even during the same transaction the sequence of execution must be respected).
+
+* Once the remaining quantity is equal to the displayed quantity, no further refresh is possible.
+The order now behaves like a normal limit order and will leave the book if it trades away completely.
+
+* For an incoming order with size larger than the total displayed quantity at a given price level, the following procedure should be followed:
+  
+  * The incoming order trades with all visible volume at the price level, whether an iceberg order or a vanilla limit order
+
+  * If there is still remaining volume in the order once all visible volume at the price level is used up, the remaining quantity is split between the non-visible components of all iceberg orders at this level according to their remaining volumes. For example if there are three iceberg orders with remaining volume 200 lots, 100 lots and 100 lots, an order for 300 lots would be split 150 to the first order and 75 to the two 100 lot orders. If the distribution doesn't divide exactly stick the extra onto the iceberg which is first in terms of time priority.
+  
+  * If there are still remaining iceberg orders at this point, refresh their volumes and continue trading. If the incoming order uses up all iceberg orders at this level, continue with any remaining volume to the next price level.
+
+
+#### Amendment
+
+* Amending the size of an iceberg order amends the total `remaining` quantity and leaves the `displayed quantity` unaffected unless the new remaining quantity is smaller than the current displayed quantity, in which case the displayed quantity is reduced to the total remaining quantity.
+
+* Amending the size/quantity of an iceberg order does not cause it to lose time priority.
+This is because the increase applies to the `remaining` quantity and not to the `displayed quantity`.
+This is allowed because the order will lose time priority on refresh, i.e. before the increased quantity is available to trade.
+
+
+#### Auctions
+
+* Icebergs can be entered or carried into auctions if the underlying TIF is supported.
+
+* Icebergs can trade in the auction uncrossing up to their current `displayed quantity` as for any other transaction that would cause a trade with an iceberg order.
```

Comment:
(hopefully the last comment): if we have an iceberg order on each side of the book and uncross only with their visible volume, they'll refresh and cross again. When in continuous trading we don't have this problem because an aggressive iceberg counts as its full volume.

Any suggestions? Do we have the notion of an "aggressor" when auction uncrossing? Does it make sense to "un-iceberg" one side?

Reply:
We don't have a notion of aggressor when uncrossing (no maker fee, remaining fees are split equally between both sides). 

I guess it makes more sense to trade the whole volume (not just visible) at uncrossing. If the party placing the iceberg doesn't want this they can mark it GFN.

Reply:
Agree whole volume. 

It's all on the price level after all. It should be split into the peaks at their position on the book and the rest at lower time priority than everything else when executing. If the remainders of multiple icebergs sitt at the same prixe and are not fully used up the traded volume should be split between them pro-rata by their total total size.

That means the algo for doing execution in an auction should be able to reuse the one for executing against passive icebergs as it's the same.